components/class-m-chart.php

```diff
@@ -601,6 +622,60 @@ public function the_content( $content ) {
 		return $this->get_chart();
 	}
 
+	/**
+	 * Hook to the m_chart_image_support filter and indicate that Chart.js supports images
+	 *
+	 * @param string $supports_images yes/no whether the library supports image generation
+	 * @param string $library the library in question
+	 *
+	 * @return string yes/no whether the library supports images
+	 */
+	public function m_chart_image_support( $supports_images, $library ) {
+		if ( $library != $this->chart_meta_fields['library'] ) {
+			return $supports_images;
+		}
+
+		return 'yes';
+	}
+
+	/**
+	 * Hook to the m_chart_instant_preview_support filter and indicate that Chart.js supports instant previews
+	 *
+	 * @param string $supports_images yes/no whether the library supports instant previews
+	 * @param string $library the library in question
+	 *
+	 * @return string yes/no whether the library supports instant previews
+	 */
+	public function m_chart_instant_preview_support( $supports_instant_preview, $library ) {
+		if ( $library != $this->chart_meta_fields['library'] ) {
+			return $supports_instant_preview;
+		}
+
+		return 'yes';
+	}
+
+	/**
+	 * Hook to the m_chart_library_class filter and return the library class if appropriate
+	 *
+	 * @param string $library_class the library class
+	 * @param string $library the library in question
+	 *
+	 * @return class the library class for this library
+	 */
+	public function m_chart_library_class( $library_class, $library ) {
+		// If Chart.js wasn't requested we'll stop here ()
+		if ( $library != $this->chart_meta_fields['library'] ) {
+			return $library_class;
+		}
+
+		if ( ! is_a( $this->library_class, 'M_Chart_Chartjs' ) ) {
```

Comment:
`is_a()` (function) is significantly slower than `instanceof` (language construct). I'd suggest switching to the latter.